test/functional/viml/function_spec.lua

```diff
@@ -1,29 +1,221 @@
+  end)
+  it('accepts weird characters', function()
+    eq('', redir_exec([[
+      function F1()
+      endfunction: }}}
+    ]]))
+    check_func('F1')
+    -- From accurev
+    eq('', redir_exec([[
+      function F2()
+      endfunction :}}}
+    ]]))
+    check_func('F2')
+    -- From cream-vimabbrev
+    eq('', redir_exec([[
+      function F3()
+      endfunction 1}}}
+    ]]))
+    check_func('F3')
+    -- From pyunit
+    eq('', redir_exec([[
+      function F4()
+      endfunction # }}}
+    ]]))
+    check_func('F4')
+    -- From vim-lldb
+    eq('', redir_exec([[
+      function F5()
+      endfunction()
+    ]]))
+    check_func('F5')
+    -- From vim-mail
+    eq('', redir_exec([[
+      function F6()
+      endfunction;
+    ]]))
+    check_func('F6')
+  end)
+  it('accepts commented bar', function()
+    eq('', redir_exec([[
+      function F1()
+      endfunction " F1 | echo 42
+    ]]))
+    check_func('F1')
+    eq('', redir_exec([[
+      function! F1()
+        return 42
+      endfunction! " F1 | echo 42
+    ]]))
+    check_func('F1', 42)
+  end)
+  it('errors out on an uncommented bar', function()
+    eq('\nE488: Trailing characters: | echo 42', redir_exec([[
+      function F1()
+      endfunction | echo 42
+    ]]))
+    check_nofunc('F1')
+  end)
+  it('allows running multiple commands', function()
+    eq('\n2', redir_exec([[
+      function F1()
+        echo 2
+      endfunction
+      call F1()
+    ]]))
+    check_func('F1', 'echo 2')
+    eq('\n2\n3\n4', redir_exec([[
+      function F2()
+        echo 2
+      endfunction F2
+      function F3()
+        echo 3
+      endfunction " F3
+      function! F4()
+        echo 4
+      endfunction!
+      call F2()
+      call F3()
+      call F4()
+    ]]))
+    check_func('F2', 'echo 2')
+    check_func('F3', 'echo 3')
+    check_func('F4', 'echo 4')
+  end)
+  it('allows running multiple commands with only one character in between',
+  function()
+    eq('\n3', redir_exec(dedent([[
+      function! F1()
+        echo 3
+      endfunction!
+      call F1()]])))
+    check_func('F1', 'echo 3', 2)
+    eq('\n4', redir_exec(dedent([[
+      function F5()
+        echo 4
+      endfunction
+      call F5()]])))
+    check_func('F5', 'echo 4', 2)
+    eq('\n5', redir_exec(dedent([[
+      function F6()
+        echo 5
+      endfunction " TEST
+      call F6()]])))
+    check_func('F6', 'echo 5', 2)
+    eq('\n6', redir_exec(dedent([[
+      function F7()
+        echo 6
+      endfunction F7
+      call F7()]])))
+    check_func('F7', 'echo 6', 2)
+    eq('\n2\n3\n4', redir_exec(dedent([[
+      function F2()
+        echo 2
+      endfunction F2
+      function F3()
+        echo 3
+      endfunction " F3
+      function! F4()
+        echo 4
+      endfunction!
+      call F2()
+      call F3()
+      call F4()]])))
+    check_func('F2', 'echo 2', 2)
+    check_func('F3', 'echo 3', 2)
+    check_func('F4', 'echo 4', 2)
   end)
 end)
+-- vim: foldmarker=▶,▲
```

Comment:
There are a number of `}}}` comments. Initially also `{{{`, but they appeared not to work. For that I made sure that foldmarkers in tests will not actually be considered fold markers by having this modeline with unicode foldmarkers I frequently use.